src/Stratis.Bitcoin.Cli/Program.cs

```diff
@@ -8,72 +9,143 @@
 using Newtonsoft.Json;
 using Stratis.Bitcoin.Configuration;
 using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.Utilities.Extensions;
 
 namespace Stratis.Bitcoin.Cli
 {
     class Program
     {
         /// <summary>
         /// The expected sequence of arguments:
-        ///    1) [network-name] [rpc-command] [rpc-params].
+        ///    1) [network-name] [options] [rpc-command] [rpc-params].
         /// OR
-        ///    2) [network-name] [api-controller "/" api-command] [api-params].
+        ///    2) [network-name] [options] [api-controller "/" api-command] [api-params].
         /// </summary>
         public static void Main(string[] args)
         {
             try
             {
+                // Preprocess the command line arguments
+                List<string> argList = new List<string>(args);
+
+                string networkName = null;
+                if (argList.Count > 0)
+                {
+                    networkName = argList.First();
+                    argList.RemoveAt(0);
+                }
+
+                List<string> optionList = new List<string>();
+                while ((argList.Count > 0) && (argList[0].StartsWith('-')))
+                {
+                    optionList.Add(argList[0]);
+                    argList.RemoveAt(0);
+                }
+
+                string command = null;
+                if (argList.Count > 0)
+                {
+                    command = argList.First();
+                    argList.RemoveAt(0);
+                }
+
+                List<string> commandArgList = new List<string>(argList);
+
                 // Display help if required.
-                if (args.Length == 0 || args.Contains("-help") || args.Contains("--help"))
+                if (command == null || argList.Contains("-help") || argList.Contains("--help"))
                 {
                     var builder = new StringBuilder();
                     builder.AppendLine("Usage:");
-                    builder.AppendLine(" dotnet run <Stratis.Bitcoin.Cli/Stratis.Bitcoin.Cli.dll> [network-name] [command] [arguments]");
+                    builder.AppendLine(" dotnet run <Stratis.Bitcoin.Cli/Stratis.Bitcoin.Cli.dll> [network-name] [options] <command> [arguments]");
```

Comment:
So do we start the CLI by executing a command directly?
It's not `dotnet run` and then we pass commands?